pyop2/compilation.py

```diff
@@ -178,8 +183,8 @@ class Compiler(ABC):
     _debugflags = ()
 
     def __init__(self, extra_compiler_flags=(), extra_linker_flags=(), cpp=False, comm=None):
-        # Get compiler version ASAP since it is used in __repr__
-        self.sniff_compiler_version()
+        # Set compiler version ASAP since it is used in __repr__
+        self.version = None
```

Comment:
This is definitely a code smell, but I don't think it's worth worrying about too much.

To address it I would personally write
```py
@cached_property
def version(self):
    # sniff compiler version...
```
so nothing then needs to be done in `__init__`.